tests_e2e/tests/ext_policy/ext_policy.py

```diff
@@ -0,0 +1,233 @@
+            }
+        self._create_policy_file(policy)
+        # # Known CRP issue - delete/uninstall operation times out instead of reporting an error.
+        # # TODO: uncomment this test case after issue is resolved
+        # # self._operation_should_fail("delete", custom_script)
+
+        # If a multiconfig extension is disallowed, no instances should be processed.
+        # RunCommand is not allowed - if we try to enable two instances, both should fail fast.
+        self._operation_should_fail("enable", run_command)
+        self._operation_should_fail("enable", run_command_2)
+
+        # If single-config extension is initially blocked, and policy is updated to allow it, extension should be
+        # successfully enabled and report status correctly.
+        self._operation_should_fail("enable", custom_script)
+        policy = \
+            {
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": False,
+                    "signatureRequired": False,
+                    "extensions": {}
+                }
+            }
+        self._create_policy_file(policy)
+        self._operation_should_succeed("enable", custom_script)
+
+    def get_ignore_error_rules(self) -> List[Dict[str, Any]]:
+        ignore_rules = [
+            #
+            # 2024-10-23T17:50:38.107793Z WARNING ExtHandler ExtHandler Dependent extension Microsoft.Azure.Monitor.AzureMonitorLinuxAgent failed or timed out, will skip processing the rest of the extensions
+            # We intentionally block extensions with policy and expect any dependent extensions to be skipped
+            {
+                'message': r"Dependent extension .* failed or timed out, will skip processing the rest of the extensions"
+            },
+            # 2024-10-23T18:01:32.247341Z ERROR ExtHandler ExtHandler Event: name=Microsoft.Azure.Monitor.AzureMonitorLinuxAgent, op=ExtensionProcessing, message=Skipping processing of extensions since execution of dependent extension Microsoft.Azure.Monitor.AzureMonitorLinuxAgent failed, duration=0
+            # We intentionally block extensions with policy and expect any dependent extensions to be skipped
+            {
+                'message': r"Skipping processing of extensions since execution of dependent extension .* failed"
+            },
+            # 2024-10-24T17:34:20.808235Z ERROR ExtHandler ExtHandler Event: name=Microsoft.Azure.Monitor.AzureMonitorLinuxAgent, op=None, message=[ExtensionPolicyError] Extension will not be processed: failed to enable extension 'Microsoft.Azure.Monitor.AzureMonitorLinuxAgent' because extension is not specified in allowlist. To enable, add extension to the allowed list in the policy file ('/etc/waagent_policy.json')., duration=0
+            # We intentionally block extensions with policy and expect this failure message
+            {
+                'message': r"Extension will not be processed"
```

Comment:
I removed the ExtensionPolicyError error class, since it never gets raised. I can add the prefix directly into the error message, if you think it's useful.